app/models/server/models/LivechatSessions.js

```diff
@@ -0,0 +1,154 @@
+import { Base } from './_Base';
+
+/**
+ * Livechat session model
+ */
+
+export class LivechatSessions extends Base {
+	constructor() {
+		super('livechat_sessions');
+	}
+
+	findOneVisitorAndUpdateSession(visitorInfo) {
+		const query = {
+			token: visitorInfo.token,
+		};
+
+		delete visitorInfo.token;
+		const update = {
+			$set: {
+				visitorInfo,
+			},
+		};
+
+		return this.update(query, update);
+	}
+
+	findOneVisitorByTokenAndUpdateStatus(token, status, chatStatus) {
+		const query = {
+			token,
+		};
+		let update;
+		const sessionInfo = this.findOne(query);
```

Comment:
All of this code needs to be process outside the model.
Here you will just update the document.

Reply:
Updating this